src/mdp/sliding_puzzle.lp

```diff
@@ -0,0 +1,71 @@
+% Sliding Puzzle
+% Another complicated planning problem
+
+% PROBLEM-SPECIFIC DOMAIN (UNINTERPRETED FUNCTIONS) ****************************
+
+%	VARIABLES:
+%	P			... describes a piece (`p1`, `p2`, ...) in the sliding puzzle.
+%	X,Y			... describes a location to put pieces on
+%					(`1,1`, `1,2`, `1,3`, ...).
+
+% 	STATE FUNCTIONS S(...):
+%	on(P,X,Y)		... describes piece `P` being on location `X,Y`.
+
+%	ACTION FUNCTIONS A(...):
+%	move(P,X,Y)	... describes the action of moving piece `P` to location `X,Y`.
+
+%	STATIC STATE (ATOMS):
+%	subgoal(P,X,Y)	... describes that `P` on position `X,Y` is
+%						(part of) the goal state.
+
+
+% KNOWLEDGE ********************************************************************
+% currentState(on(p1,1,1)).
+% currentState(on(p0,1,0)).
+% subgoal(p1,0,1).
+% subgoal(p1,0,1). subgoal(p0,0,0).
+
+% Static knowledge
+piece(P) :- tic(on(P,_,_), 0).
+
+size(2..4).
```

Comment:
Why is the size here fixed? I thought puzzle size is provided by the python code?

Reply:
forgot this!
butr already fixed it